src/iso9660.rs

```diff
@@ -107,31 +107,41 @@ impl IsoFs {
                 let c = c.to_str().unwrap(); // `path` is &str
                 dir = self
                     .get_dir_record(&dir, c)?
-                    .ok_or_else(|| anyhow!("intermediate directory {} does not exist", c))?
+                    .ok_or_else(|| {
+                        NotFound(format!("intermediate directory {} does not exist", c))
+                    })?
                     .try_into_dir()
                     .map_err(|_| {
-                        anyhow!("component {:?} in path {} is not a directory", c, path)
+                        NotFound(format!(
+                            "component {:?} in path {} is not a directory",
+                            c, path
+                        ))
                     })?;
             } else {
                 bail!("path is not canonical: {}", path);
             }
         }
 
         self.get_dir_record(&dir, filename)?.ok_or_else(|| {
-            anyhow!(
+            anyhow!(NotFound(format!(
                 "no record for {} in directory {}",
                 filename,
                 parent_dir.display()
-            )
+            )))
         })
     }
 
+    /// Returns the record for a specific path if it exists.
+    pub fn try_get_path(&mut self, path: &str) -> Result<Option<DirectoryRecord>> {
```

Comment:
Hmm, could we instead just make `NotFound` public and clients of `get_path` can just look for it if needed?